src/couch_views/src/couch_views_encoding.erl

```diff
@@ -0,0 +1,105 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(couch_views_encoding).
```

Comment:
If I understand correctly this module is a replacement for `couch_ejson_compare.erl`. The `couch_ejson_compare.erl` is used from other applications as well (mango and fabric). Should it be more generic encoding module?

Feel free to ignore.

Reply:
This can definitely be used in mango. How would we make more generic? Everything here would be used the same for any other index

Reply:
> How would we make more generic?

We could rename the module and move it into another app. I think it could be in one of:
- `couch` - might not be a good fit because we would have to add `erlfdb` into list of applications in `couch.app.src`
- `couch_index` - it is a generic abstraction to work with all index types. Might not be a good fit because Paul considers introduction of couch_index a bad move. 
- `fabric` - could be a good place since it already includes `erlfdb`. 

Feel free to ignore this proposal. It is good enough for the first version. We can move it latter when we start using it from mango.

Reply:
I like the ideas. Lets see how Mango goes and then we can look to change this.